src/gluonts/mx/trainer/_base.py

```diff
@@ -251,20 +250,18 @@ def loop(
                         self.avg_strategy.load_averaged_model(net)
 
                     with tqdm(batch_iter) as it:
-                        for batch_no, data_entry in enumerate(it, start=1):
+                        for batch_no, batch in enumerate(it, start=1):
                             if self.halt:
                                 break
 
-                            inputs = [data_entry[k] for k in input_names]
-
                             if first_forward:
                                 first_forward = False
-                                _ = net(*inputs)
+                                _ = net(*batch.values())
```

Comment:
We now hide what a batch is, maybe we want to add a line here in the trainer that it is a dictionary with different fields?

Reply:
yes, makes sense to have a comment for clarity I think